src/module/skb_tracking/skb_tracking.rs

```diff
@@ -51,6 +53,16 @@ impl Collector for SkbTrackingCollector {
     fn init(&mut self, _: &CliConfig, probes: &mut ProbeManager) -> Result<()> {
         self.init_tracking(probes)
     }
+
+    fn stop(&mut self) -> Result<()> {
+        match self.garbage_collector.take() {
+            Some(gc) => {
+                self.state.terminate();
+                gc.join().or_else(|_| bail!("failed to stop gc"))
```

Comment:
I guess if we add named threads (which is trivial and not part of the series), adding the thread name as well could make sense.

Reply:
fyi, in #113 I have proposed refactoring the garbage collector thread into it's own struct which which includes setting the name. It will probably also help joining the threads better.